enhancements/baremetal/bootstrap-kubelet-ip.md

```diff
@@ -0,0 +1,132 @@
+---
+title: bootstrap-kubelet-ip
+authors:
+  - "@tsorya"
+reviewers:
+  - "@hardys"
+  - "@dtantsur"
+  - "@dhellmann"
+  - "@patrickdillon"
+approvers:
+  - "@hardys"
+  - "@dtantsur"
+  - "@dhellmann"
+  - "@patrickdillon"
+api-approvers:
+  - None
+creation-date: 2022-07-10
+last-updated: 2022-07-10
+tracking-link:
+  - https://issues.redhat.com/browse/MGMT-11102
+see-also:
+  - https://bugzilla.redhat.com/show_bug.cgi?id=2070318
+replaces:
+  - https://github.com/openshift/installer/pull/6042/files
+---
+
+# Bootstrap External IP
+
+## Summary
+When installing a new cluster with UPI, you can set machine networks that will 
+be set to `networkConfig` field in the `install-config.yaml` file to configure the control plane network
+interfaces for the cluster hosts. However, users cannot configure 
+the machine network for the bootstrap node.
+
+## Motivation
+
+In environments where bootstrap machine has more than one ip address, we should be able to set which ip that 
+will be configured as a hostIp in kubelet. This ip will be used in bootstrap kube-api service as advertised ip.
+This will allow traffic to go through expected interface in case where we have more than one interface and kubelet on bootstrap 
+choose ip different from subnet configured on masters.
+The user can work around this by modifying the bootstrap ignition file; however,
+this isn't a friendly experience.
+
+### User Stories
+
+- As a user, I want to be able to specify host ip address for bootstrap machine
+
+### Goals
+
+- Support cluster installation in environments with bootstrap machine with more than one ip configured
+- Improve the level of customization of bootstrap networking
+
+### Non-Goals
+
+- Provide means for configuring every aspect of bootstrap networking
+
+## Proposal
+
+We will add a new field, `bootstrapNodeIP` to allow for further customization of
```

Comment:
It is possible, added it as alternative solution. The issue is that it will require adding service that should read networking.machineNetwork inside bootstrap and find relevant nic with ip that is part of given subnet. It should run before kubelet and set env file for example.
Was not sure if it will be something easy to do in bootstrap flow (not enough knowledge on it) but if you think it is better, we can go for it.